packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -82,46 +81,47 @@ The POC is found in this branch [Worker Rendering POC](https://github.com/ni/nim
-    // the highlight state of each die as a matrix; user will have to pre-calculate tags into highlighted conditions.
-    dieHighlightsLayer: Int8Array;
-    // a 32 bitset array of tags for each die; aligns more closely to the existing public api but limits users to 32 tags.
-    dieHighlightsLayer: Int32Array;
-    // metadata array for each die; it will not be sent to the worker
-    metadata : unknown[]
+import { Table } from 'apache-arrow';
+
+export class WaferMap extends FoundationElement {
+...
+public diesTable: Table | undefined;
+public highlightedTable: Table | undefined;
+...
 }
 ```
 
-Using TypedArrays has the benefit of direct transfer to web workers without structured cloning of the object by transferring the arrayBuffers and reconstructing the object. Other benefits of typedArrays include the low access time when iterating over the values, more memory efficiency and faster direct access to metadata layers values. The previous inputs can be adapted to this new structure to maintain backwards compatibility.
+It will be using the [Apache Arrow Table](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html).
+It will require at least three columns for the `diesTable`:
 
-This API will have [optimized byte-array interop from Blazor](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/byte-array-interop) and should be supported by Angular as a [vanilla javascript feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
+-   The row and column indices will be `Int32` columns
+-   The values will be a `Float32` column.
 
-The alternative to the above mentioned data structure is an [apache arrow](https://arrow.apache.org/docs/js/index.html) table with columns and metadata.
+They will be checked at runtime and a `WaferMapValidity` flag will be raised signaling an `invalidTableInput`.
 
-Pros of using Apache Arrow:
+The schema will be extensible. This will induce a breaking change in the API, as the metadata which was previously `unknown` will not recorded in table, but the hover event will reference an index which can e used by the client to select the metadata outside the component.
 
--   A row based format that aligns well with the existing public api
--   Well supported and tested format
--   Nice public API to use, we don't have to invent a new format, just document our schema for the arrow tables
--   Designed for large dataset visualizations
+The `highlightedTable` will have the same columns, but they will contain rows only partially filled with values, which will be used to filter the `diesTable` and enable highlighting. The values which are not empty on each individual row, including `colIndex`, `rowIndex`, `value` and others will be used to filter the table as an `AND` operation. Multiple rows will be used as filters with the `OR` operation. More details regarding highlights will be discussed in an open issue.
```

Comment:
As I mentioned in the [Teams chat](https://teams.microsoft.com/l/message/19:meeting_OTJkYmU1M2QtN2Q2YS00M2E5LTgzODctMzEyZTkxNTA0ZDFi@thread.v2/1709071000119?context=%7B%22contextType%22%3A%22chat%22%7D), it appears that the Apache Arrow Table structure supports list (array) types. This is suggestive that we could support the current API (`string[]`) for `highlightedTags`, which we feel would be preferable, barring some other noted reason why the API change is beneficial.

With that said, this determination can happen at a later time, if it is worthwhile to put this decision on the backburner, to unblock the remainder of the work outlined here. If you do decide to make this an open issue, then I believe what is proper is to remove all mention of the `highlightedTable` from this section and the above "Public API" description.

@jattasNI @rajsite